static/app/utils/performance/contexts/pageAlert.tsx

```diff
@@ -0,0 +1,81 @@
+import {createContext, Fragment, useContext, useState} from 'react';
+import {Theme} from '@emotion/react';
+
+import {Alert} from 'sentry/components/alert';
+
+export type PageAlertOptions = {
+  message: React.ReactNode | undefined;
+  type: keyof Theme['alert'];
+};
+
+const pageErrorContext = createContext<{
+  setPageError: (message: string | undefined) => void;
+  setPageInfo: (message: string | undefined) => void;
+  setPageMuted: (message: string | undefined) => void;
+  setPageSuccess: (message: string | undefined) => void;
+  setPageWarning: (message: string | undefined) => void;
+  pageAlert?: PageAlertOptions;
+}>({
+  pageAlert: undefined,
+  setPageError: (_: string | undefined) => {},
+  setPageInfo: (_: string | undefined) => {},
+  setPageMuted: (_: string | undefined) => {},
+  setPageSuccess: (_: string | undefined) => {},
+  setPageWarning: (_: string | undefined) => {},
+});
```

Comment:
The types here are a little funky! At a minimum, is it possible to extract `(_: string | undefined) => {}` and `(message: string | undefined) => void` into a type?

Reply:
Also, these should accept a `React.ReactNode` to match the component, no?